examples/notebooks/03_model_scoring.py

```diff
@@ -0,0 +1,220 @@
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# %% [markdown]
+"""
+# Scoring Models in Earth-2 MIP
+
+The following notebook will demonstrate how to use Earth-2 MIP to perform a scoring
+workflow to assess the accuracy of AI models using ERA5 reanalysis data as the
+ground truth. This can then be extended to score custom models placed into the model
+registry. This tutorial also covers details about using a HDF5 datasource, the expected
+format this data should be formatted and how to use H5 files for evaluating models over
+a year's worth of data.
+
+In summary this notebook will cover the following topics:
+
+- Implementing a basic scoring workflow in Earth-2 MIP
+- HDF5 datasource and the expected data format of the H5 files
+
+"""
+# %%
+import os
+import xarray
+import datetime
+import numpy as np
+
+# %% [markdown]
+"""
+## Setting up HDF5 data
+
+The first step of scoring is handling the target data. One could simply use the
+CDSDatasource to download target data on the fly, but depending on how comprehensive
+the scoring is, this can prove quite slow.
+Additionally, many scoring pipelines require on-prem data.
+Thus, this will demonstrate how to use the HDF5 datasource.
+
+The HDF5 data source assumes that the data to be loaded is stored in the general form:
+
+year.h5
+ | - field (time, channels, grid)
+
+For AFNO which requires 34 channels with a time-step size of 6 hours, an H5 file will
+have the following form of data:
+
+2017.h5
+ | - field (1460, 34, 720, 1440)
+2016.h5
+ | - field (1464, 34, 720, 1440)
+2015.h5
+ | - field (1460, 34, 720, 1440)
+
+(Note the later two dims have some flexibility with regridding)
+
+One option to build these H5 files from scratch is to use the ERA5 mirror scripts
+provided in [Modulus](https://github.com/NVIDIA/modulus/tree/main/examples/weather/dataset_download).
+For the rest of this tutorial, it is assumed that 2017.h5 is present for the full year.
+"""  # noqa: E501
+
+# %%
+h5_folder = "/mount/34vars/test/"
+
+# %% [markdown]
+"""
+## HDF5 Datasource
+
+With the H5 files properly formatted, a H5 datasource can now get defined.
+This requires two items: a root directory location of the H5 files as well as some
+metadata.
+The metadata is a JSON/dictionary object that helps Earth-2 MIP index the H5 file.
+Metadata for a 34 channel H5 files described above is given below.
```

Comment:
Maybe mention that it should be read from data.json if using the hdf5.DataSource.from_path constructor.

Reply:
Good call, added